flink-table/flink-table-common/src/main/java/org/apache/flink/table/plan/stats/ColumnStats.java

```diff
@@ -217,6 +219,55 @@ public ColumnStats copy() {
 		}
 	}
 
+	@SuppressWarnings("unchecked")
+	public ColumnStats merge(ColumnStats other) {
+		Long ndv = Stream.of(this.ndv, other.ndv).filter(Objects::nonNull).reduce(Long::sum).orElse(null);
```

Comment:
i think the strategy should be: if one of the `ndv`s is null, the merged `ndv` should also be null. for example: there are many `ColumnStats`s to merge, and only a few of `ColumnStats`s have non-null `ndv` values, the result of current strategy is an untrusted value.